services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -201,6 +204,53 @@ class HakemusServiceITest(
         }
     }
 
+    @Nested
+    inner class GetWithPaatokset {
+        @Test
+        fun `throws an exception when the hakemus does not exist`() {
+            val failure = assertFailure { hakemusService.getWithPaatokset(1234) }
+
+            failure.all {
+                hasClass(HakemusNotFoundException::class)
+                messageContains("id 1234")
+            }
+        }
+
+        @Test
+        fun `returns hakemus`() {
+            val hakemus = hakemusFactory.builder(USERNAME).withMandatoryFields().save()
+
+            val response = hakemusService.getWithPaatokset(hakemus.id)
+
+            assertThat(response.hakemus).all {
+                prop(Hakemus::id).isEqualTo(hakemus.id)
+                prop(Hakemus::applicationIdentifier).isNull()
+                prop(Hakemus::hankeId).isEqualTo(hakemus.hankeId)
+                prop(Hakemus::id).isEqualTo(hakemus.id)
+                prop(Hakemus::applicationData).isInstanceOf(JohtoselvityshakemusData::class).all {
+                    prop(JohtoselvityshakemusData::name).isEqualTo(hakemus.applicationData.name)
+                }
+            }
+            assertThat(response.paatokset).isEmpty()
+        }
+
+        @Test
+        fun `returns paatokset when they exist`() {
+            val hakemus =
+                hakemusFactory
+                    .builder()
+                    .withMandatoryFields()
+                    .withStatus(ApplicationStatus.DECISION)
+                    .save()
+            val paatos1 = paatosFactory.save(hakemus, tila = PaatosTila.KORVATTU)
+            val paatos2 = paatosFactory.save(hakemus, hakemus.applicationIdentifier + "-1")
+
+            val response = hakemusService.getWithPaatokset(hakemus.id)
+
+            assertThat(response.paatokset).containsExactlyInAnyOrder(paatos1, paatos2)
```

Comment:
I was wondering whether the decisions should be sorted in the backend and returned in correct order?

Reply:
Good question. Maybe the order is best to leave for the UI to decice how it presents e.g. the decision links.